src/flamenco/runtime/program/fd_bpf_loader_v3_program.c

```diff
@@ -815,9 +815,13 @@ process_loader_upgradeable_instruction( fd_exec_instr_ctx_t * instr_ctx ) {
       return FD_EXECUTOR_INSTR_ERR_ACC_DATA_TOO_SMALL;
     }
 
-    uchar * programdata_data = buffer->data + buffer_data_offset;
-    ulong   programdata_size = fd_ulong_sat_add( SIZE_OF_PROGRAM, programdata_len );
+    uchar * programdata_data = buffer->data    + buffer_data_offset;
+    ulong   programdata_size = programdata_len - buffer_data_offset;
 
+    /* We use programdata_size instead of UpgradeableLoaderState::size_of_program().saturating_add(programdata_len)
+       because that is the length of our buffer, and that argument to deploy_program (account_size) in the rust code is not used
+       in our version of deploy_program because we do not have a program cache
+       https://github.com/anza-xyz/agave/blob/da6f7f2d317ed978701b08c3cc3e8d62756e565a/programs/bpf_loader/src/lib.rs#L146 */
```

Comment:
Need to rewrite this to reflect changes above

Reply:
moved comment to top of deploy program